### PR TITLE
Fix blacklist bypass by blocking `qbot`

### LINKED ISSUE #56: Blocking `qbot` allows user to send blacklisted text
**Explanation and cause:**
This is because `qbot` sends the direct message to the user _before_ deleting the message ([here](https://github.com/arch-community/qbot/blob/main/modules/blacklist.rb#L44) on line 44 of the `blacklist` module), if the user blocks `qbot`, it will cause an exception preventing the deletion of the message.

**Steps to reproduce:**
1. Block `qbot` on discord
2. Send a message with blacklisted text in it

**Expected result:**
`qbot` deletes the message

**Actual result:**
Nothing happens (presumably, `qbot` raises an exception)

**Proposed fix:**
Have `qbot` delete the message prior to sending the direct message

### PR DESCRIPTION
**Pull request by FjorgeTheHuman**
Closes #56.

I didn't add a translation for the `en` locale because I don't know what to put there.

P.S. I can't figure out how to not have the second Fjorge when I `git push`

### CODE REVIEW

**Comment by thewarsawpakt** [comment]
> P.S. I can't figure out how to not have the second Fjorge when I git push

You can try github ssh authentication.

**Comment by FjorgeTheHuman** [comment]
> You can try github ssh authentication.

I tried that and I don't think it worked in the past, unless we're talking about different things.

**Comment by thewarsawpakt** [comment]
> I tried that and I don't think it worked in the past, unless we're talking about different things.

I've used [this](https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh) article recently to get it set up.

**Comment by FjorgeTheHuman** [comment]
> I've used [this](https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh) article recently to get it set up.

I'll take a look later, thanks!

File: modules/blacklist.rb [hunk 1]
```diff
@@ -41,7 +41,12 @@ module Blacklist
        .first
 
   if bl
-    event.message.author.pm t('blacklist.message', bl.re, contents)
     event.message.delete
+    begin
+      event.message.author.pm t('blacklist.message', bl.re, contents)
+    rescue Discordrb::Errors::NoPermission
+      # Catch the nopermission error and ask for an unblock
+      event.respond t('blacklist.unblock', format('<@%s>', event.message.author.id))
```
Comment (anna328p) [inline modules/blacklist.rb:49]: Please use `Discordrb::User#mention`

**Review by anna328p — CHANGES_REQUESTED** [review]

**Pull request merged**